src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	public class MathExpressionConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		/// Calculate the incoming expression string with one variable.
+		/// </summary>
+		/// <param name="value">The variable X for an expression</param>
+		/// <param name="targetType">The type of the binding target property. This is not implemented.</param>
+		/// <param name="parameter">The expression to calculate.</param>
+		/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
+		/// <returns>A <see cref="double"/> The result of calculating an expression.</returns>
+		public object? Convert(object? value, Type? targetType, object? parameter, CultureInfo culture)
+		{
+			if (parameter is not string expression)
+				throw new ArgumentException("The parameter should be of type String");
+
+			if (value is null || !double.TryParse(value.ToString(), out var xValue))
```

Comment:
We don't use `pattern matching` for checking null.
```suggestion
			if (value == null || !double.TryParse(value.ToString(), out var xValue))
```